packages/nvidia_nat_mcp/tests/client/test_mcp_client_impl.py

```
@@ -174,6 +175,22 @@ def _make_group(server_cfg=None, client_cfg=None):
 
 
+@patch("nat.builder.context.Context")
+def test_resolve_user_id_from_auth_header(mock_context_cls):
+    """When auth_token_header is configured, user ID is resolved from JWT claims."""
+    token: str = jwt.encode({"name": "Alice", "sub": "alice-123"}, "a" * 32, algorithm="HS256")
+    ctx = MagicMock()
+    ctx.metadata.headers = {"Authorization": f"Bearer {token}"}
+    mock_context_cls.get.return_value = ctx
+
+    server_cfg = MCPServerConfig(transport="streamable-http", url="http://localhost:9901/mcp")
+    client_cfg = MCPClientConfig(server=server_cfg, auth_token_header="Authorization")
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to argument: "auth_token_header"

(S106)